nixos/modules/services/security/crowdsec.nix

```diff
@@ -0,0 +1,867 @@
+        fi
+        $sudo ${lib.getExe' cfg.package "cscli"} -c=${configFile} "$@"
+      '';
+
+      localScenariosMap = (map (format.generate "scenario.yaml") cfg.localConfig.scenarios);
+      localParsersS00RawMap = (
+        map (format.generate "parsers-s00-raw.yaml") cfg.localConfig.parsers.s00Raw
+      );
+      localParsersS01ParseMap = (
+        map (format.generate "parsers-s01-parse.yaml") cfg.localConfig.parsers.s01Parse
+      );
+      localParsersS02EnrichMap = (
+        map (format.generate "parsers-s02-enrich.yaml") cfg.localConfig.parsers.s02Enrich
+      );
+      localPostOverflowsS01WhitelistMap = (
+        map (format.generate "postoverflows-s01-whitelist.yaml") cfg.localConfig.postOverflows.s01Whitelist
+      );
+      localContextsMap = (map (format.generate "context.yaml") cfg.localConfig.contexts);
+      localNotificationsMap = (map (format.generate "notification.yaml") cfg.localConfig.notifications);
+      localProfilesFile = pkgs.writeText "local_profiles.yaml" ''
+        ---
+        ${lib.strings.concatMapStringsSep "\n---\n" builtins.toJSON cfg.localConfig.profiles}
+        ---
+      '';
+      localAcquisisionFile = pkgs.writeText "local_acquisisions.yaml" ''
+        ---
+        ${lib.strings.concatMapStringsSep "\n---\n" builtins.toJSON cfg.localConfig.acquisitions}
+        ---
+      '';
+
+    in
+    lib.mkIf (cfg.enable) {
+
+      assertions = [
+        {
+          assertion = lib.trivial.xor (cfg.settings.lapi.credentials != null) (
+            cfg.settings.lapi.credentialsFile != null
+          );
+          message = "Please specify either services.crowdsec.settings.lapi.credentialsFile or services.crowdsec.settings.lapi.credentials, not more, not less.";
+        }
+      ];
+
+      warnings =
+        [ ]
+        ++ (
+          if cfg.localConfig.profiles == [ ] then
+            [
+              "By not specifying profiles in services.crowdsec.localConfig.profiles, CrowdSec will not react to any alert by default"
+            ]
+          else
+            [ ]
+        )
+        ++ (
+          if cfg.localConfig.acquisitions == [ ] then
+            [
+              "By not specifying acquisitions in services.crowdsec.localConfig.acquisitions, CrowdSec will not look for any data source"
+            ]
+          else
+            [ ]
+        );
+
+      services.crowdsec.settings.general = with lib; {
+        common = {
+          daemonize = mkDefault false;
+          log_media = mkDefault "stdout";
+        };
+        config_paths = {
+          config_dir = mkDefault confDir;
+          data_dir = mkDefault stateDir;
+          simulation_path = mkDefault simulationFile;
+          hub_dir = mkDefault hubDir;
+          index_path = mkDefault (lib.strings.normalizePath "${stateDir}/hub/.index.json");
+          notification_dir = mkDefault notificationsDir;
+          plugin_dir = mkDefault pluginDir;
+          pattern_dir = mkDefault patternsDir;
+        };
```

Comment:
I don't think we really support changing those, so most of them should not be using mkDefault